### ISSUE TITLE
InteractiveTable: Support specific column widths

### ISSUE THREAD

**Issue by tskarhed** [grafana/grafana#80980]
![image](https://github.com/grafana/grafana/assets/1438972/7c8c44af-9775-4a16-a65c-15903b4f1fef)

I find the column spacing and/or width to be confusing. There should either be some guidelines around this, or some technical fix.

In the first case for `LDAP Connection`, the first column has `disableGrow: true`, because otherwise the port would end up in the middle, which does not make any sense. The connection status icon ends up on the completely other side. The quick fix for this would be to set a fixed/max width to the entire table.

Also, how should tables like the two bottom ones be managed? They have similar columns, and it would look nice if both could be aligned. Adding a title above these might be enough to separate them more.

**Comment by Clarity-89** [comment]
To the best of my knowledge, there's no way to set column widths via the props, the best you could do is to wrap each cell into a div with a specified width: 
```tsx
      {
        id: 'host',
        header: 'Host',
        disableGrow: true,
        cell: (host: CellProps<any>) => <div style={{ width: 200 }}>{host.cell.value}</div>,
      },
```
We'd probably look into making specifying the column widths easier.

**Comment by JoaoSilvaGrafana** [comment]
Yep I don't think this is possible at the moment, it would be nice to have a width prop per column with either using some design tokens or some predefined values for small/medium/large or something. I'll move this to our board 👍

**Comment by DivyamUp14** [comment]
Hi @tskarhed , I've implemented this! I opened a PR that adds an optional width property to the Column interface (supporting both px and %), while maintaining full backward compatibility. You can check out the details in the above PR.

<img width="1176" height="546" alt="Image" src="https://github.com/user-attachments/assets/d213ff3d-4b0b-4f5b-b11a-2c05764b808f" />
